test/e2e/suite/conformance/certificates/tests.go

```diff
@@ -694,6 +694,50 @@ func (s *Suite) Define() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
+		s.it(f, "Creating a Gateway with annotations for issuerRef and other Certificate fields", func(issuerRef cmmeta.ObjectReference) {
+			name := "testcert-gateway"
+			secretName := "testcert-gateway-tls"
+			domain := e2eutil.RandomSubdomain(s.DomainSuffix)
+			duration := time.Hour * 999
+			renewBefore := time.Hour * 111
+
+			By("Creating a Gateway with annotations for issuerRef and other Certificate fields")
+			gw, route := e2eutil.NewGateway(name, f.Namespace.Name, secretName, map[string]string{
+				"cert-manager.io/issuer":       issuerRef.Name,
+				"cert-manager.io/issuer-kind":  issuerRef.Kind,
+				"cert-manager.io/issuer-group": issuerRef.Group,
+				"cert-manager.io/common-name":  domain,
+				"cert-manager.io/duration":     duration.String(),
+				"cert-manager.io/renew-before": renewBefore.String(),
+			}, domain)
+
+			gw, err := f.GWClientSet.NetworkingV1alpha1().Gateways(f.Namespace.Name).Create(context.TODO(), gw, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			_, err = f.GWClientSet.NetworkingV1alpha1().HTTPRoutes(f.Namespace.Name).Create(context.TODO(), route, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// XXX(Mael): the CertificateRef seems to contain the Gateway name
+			// "testcert-gateway" instead of the secretName
+			// "testcert-gateway-tls".
+			certName := gw.Spec.Listeners[0].TLS.CertificateRef.Name
+
+			By("Waiting for the Certificate to exist...")
+			Expect(e2eutil.WaitForCertificateToExist(
+				f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name), certName, time.Minute,
+			)).NotTo(HaveOccurred())
+
+			// Verify that the ingres-shim has translated all the supplied
+			// annotations into equivalent Certificate field values
+			By("Validating the created Certificate")
+			cert, err := f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name).Get(context.TODO(), certName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(cert.Spec.DNSNames).To(ConsistOf(domain))
+			Expect(cert.Spec.CommonName).To(Equal(domain))
+			Expect(cert.Spec.Duration.Duration).To(Equal(duration))
+			Expect(cert.Spec.RenewBefore.Duration).To(Equal(renewBefore))
```

Comment:
We don't seem to test that the Gateway actually uses that certificate.

I think ideally we should test that it's used- if we don't do that, we could skip installing istio and just install the Gateway CRDs?

Reply:
You are right, installing istio was a bit premature in this PR 😞  I'll remove it for now.

Reply:
Ahh, I didn't realize that we were relying on istio for the GatewayClass creation?
But I guess the Gateway gets applied anyway and cert-manager does not check if the Gateway is ready.

Reply:
I assumed I would create the GatewayClass manually (since `istioctl` does not install the Gateway and GatewayClass I think).

Yes, the Gateway gets applied even if there is no GatewayClass (that's why I did not bother creating a GatewayClass for istio).

Reply:
I guess the code can be re-used in the http-01 solver PR, if that's to be tested with istio? (for the gateway)